### PR TITLE
Hide progress on non-ttys

### PR DESCRIPTION
**Pull request by mvz**
Implements #1118. Like #1119, but targeting master.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -88,13 +89,43 @@ class Alfa
...
+Reek uses [Cucumber](https://cucumber.io/) with
+[Aruba](https://github.com/cucumber/aruba) for integration tests. Keep the
+following in mind when writing cucumber features.
+
+#### What to test
+
+Not everything needs a cucumber feature. We try to limit cucumber features to
+things that really require end-to-end testing of Reek's behavior. In
+particular, this means individual smell detectors should not have their own
+scenarios.
```
Comment (troessner) [inline CONTRIBUTING.md]: 👍

File: lib/reek/report/text_report.rb [hunk 2]
```diff
@@ -13,19 +13,19 @@ class TextReport < BaseReport
       def initialize(*args)
         super(*args)
 
-        progress_formatter.header
+        print progress_formatter.header
```
Comment (troessner) [inline lib/reek/report/text_report.rb]: Yep, big 👍  here and below for this change.

**Comment by troessner** [comment]
Please rebase and squash and then I'm happy to merge this one!

**Comment by mvz** [comment]
Squashed and updated to adjust todo list feature.

**Comment by troessner** [comment]
And merged it is! Care for a release @mvz ? :)

**Pull request merged**